### PR TITLE
Add list of default MMLU tasks as a constant

### PR DESCRIPTION
**Pull request by nathan-weinberg**
@danmcp @cdoern and I met earlier and discussed moving the default MMLU tasks from the CLI into the Library. Motivation is that the tasks are really a core part of MMLU and generally shouldn't be overriden. I've left provisions to allow for overriding, but this way the CLI as well as any other library consumers will be using these tasks as a default.

### CODE REVIEW

**Comment by nathan-weinberg** [comment]
@alimaredia has highlighted that a tuple may not be compatible with MMLU - need to test this

**Review by danmcp — APPROVED** [review]
LGTM

**Comment by nathan-weinberg** [comment]
Tested, seems the tuple is fine
```shell
(venv) [ec2-user@ip-10-0-14-244 eval (mmlu-tasks)]$ python3.11 test_mmlu.py 
2024-06-26:20:53:13,736 INFO     [evaluator.py:131] Setting random seed to 0 | Setting numpy seed to 1234 | Setting torch manual seed to 1234
2024-06-26:20:53:13,776 INFO     [huggingface.py:162] Using device 'cuda'
/home/ec2-user/nweinber/venv/lib/python3.11/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
Loading checkpoint shards: 100%|████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:05<00:00,  1.83s/it]
You are using the default legacy behaviour of the <class 'transformers.models.llama.tokenization_llama_fast.LlamaTokenizerFast'>. This is expected, and simply means that the `legacy` (previous) behavior will be used so nothing changes for you. If you want to use the new behaviour, set `legacy=False`. This should only be set if you understand what it means, and thoroughly read the reason why this was added as explained in https://github.com/huggingface/transformers/pull/24565
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
2024-06-26:20:53:19,764 WARNING  [task.py:322] [Task: tonsils] has_training_docs and has_validation_docs are False, using test_docs as fewshot_docs but this is not recommended.
2024-06-26:20:53:19,765 WARNING  [task.py:322] [Task: tonsils] has_training_docs and has_validation_docs are False, using test_docs as fewshot_docs but this is not recommended.
2024-06-26:20:53:19,844 WARNING  [task.py:322] [Task: softball] has_training_docs and has_validation_docs are False, using test_docs as fewshot_docs but this is not recommended.
2024-06-26:20:53:19,845 WARNING  [task.py:322] [Task: softball] has_training_docs and has_validation_docs are False, using test_docs as fewshot_docs but this is not recommended.
2024-06-26:20:53:19,849 WARNING  [evaluator.py:222] Overwriting default num_fewshot of softball from None to 2
2024-06-26:20:53:19,849 WARNING  [evaluator.py:222] Overwriting default num_fewshot of tonsils from None to 2
2024-06-26:20:53:19,850 INFO     [task.py:395] Building contexts for softball on rank 0...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 24/24 [00:00<00:00, 169.24it/s]
2024-06-26:20:53:19,995 INFO     [task.py:395] Building contexts for tonsils on rank 0...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 27/27 [00:00<00:00, 175.92it/s]
2024-06-26:20:53:20,153 INFO     [evaluator.py:362] Running loglikelihood requests
Running loglikelihood requests: 100%|███████████████████████████████████████████████████████████████████████████████████████| 204/204 [00:03<00:00, 53.56it/s]
0.7450980392156863
{'mmlu_pr': {'score': 0.7450980392156863, 'stderr': 0.0619412614751624}}
```

**Comment by nathan-weinberg** [comment]
I want to do a couple more checks before merging

File: src/instructlab/eval/mmlu.py [hunk 1]
```diff
@@ -27,7 +91,7 @@ class MMLUEvaluator(Evaluator):
     def __init__(
         self,
         model_path,
-        tasks: list[str],
+        tasks: tuple[str, ...] = MMLU_TASKS,
```
Comment (JamesKunstle) [inline src/instructlab/eval/mmlu.py:94]: is that the type-annotation for "a tuple of one or more strings"? I haven't seen that before

**Comment by JamesKunstle** [comment]
@nathan-weinberg that's actually super funny, I requested that with #25 and implemented a facsimile for that in my draft PR in the cli.

**Review by JamesKunstle — APPROVED** [review]

File: src/instructlab/eval/mmlu.py (hunk 1 above)
Comment (nathan-weinberg) [reply src/instructlab/eval/mmlu.py:94]: Yessir it is

**Comment by nathan-weinberg** [comment]
After more testing, it does indeed need to be a list not a tuple
```shell
test1 failed: Expected a 'str' or 'list' but received <class 'tuple'>.
```
Will push up a fix soon

**Comment by JamesKunstle** [comment]
@nathan-weinberg can this be merged? it looks good to me

**Comment by nathan-weinberg** [comment]
Given the prior approvals I think so - if anyone has concerns with the lint rule changes let me know!

**Pull request merged**